plugin/adsk/scripts/mayaUsdMenu.py

```diff
@@ -0,0 +1,168 @@
+from maya import cmds
+
+import ufe
+import usdUfe
+import mayaUsd.ufe
+
+from collections import defaultdict
+from functools import partial
+
+from mayaUSDRegisterStrings import getMayaUsdString
+
+_pluginNiceNames = {
+    "usdGeom": "Geometry",
+    "usdLux": "Lighting",
+    "mayaUsd_Schemas": "Maya Reference",
+    "usdMedia": "Media",
+    "usdRender": "Render",
+    "usdRi": "RenderMan",
+    "usdShade": "Shading",
+    "usdSkel": "Skeleton",
+    "usdUI": "UI",
+    "usdVol": "Volumes",
+    "usdProc": "Procedural",
+    "usdPhysics": "Physics",
+    "usdArnold": "Arnold",
+    # Skip legacy AL schemas
+    "AL_USDMayaSchemasTest": "",
+    "AL_USDMayaSchemas": "",
+}
+
+def _getNicePluginName(pluginName):
+    '''
+    Convert the plugin name to a nicer name we designed for users.
+    Return an empty name for plugin we don't want to show.
+    '''
+    return _pluginNiceNames.get(pluginName, pluginName)
+
+
+def _groupSchemasByPlugins(schemas):
+    '''
+    Group schemas by the plugin name that contains them and then
+    by the schema type name, giving a boolean flag if the schema
+    is a multi-apply schema.
+    '''
+    schemaByPlugins = defaultdict(dict)
+    for info in schemas:
+        pluginName = _getNicePluginName(info['pluginName'])
+        if not pluginName:
+            continue
+        schemaByPlugins[pluginName][info['schemaTypeName']] = info['isMultiApply']
+    return schemaByPlugins
+
+
+def _getUsdItemsInSelection():
+    '''
+    Retrieve the UFE items in the selection that are USD prims.
+    '''
+    ufeSelection = ufe.GlobalSelection.get()
+    return [item for item in ufeSelection if item.runTimeId() == mayaUsd.ufe.getUsdRunTimeId()]
+
+
+def _askForInstanceName(prettyTypeName):
+    '''
+    Ask the user for the multi-apply schema instance name.
+    '''
+    title = getMayaUsdString("kAddSchemaMenuItem")
+    message = getMayaUsdString("kAddSchemaInstanceMesage") % prettyTypeName
+
+    result = cmds.promptDialog(
+        message=message, title=title,
+        button=('OK', 'Cancel'),
+        defaultButton='OK', cancelButton='Cancel')
+    if result != 'OK':
+        return ''
+
+    return cmds.promptDialog(query=True, text=True)
+
+
+def _applySchemaMenuItemCallback(prettyTypeName, schemaTypeName, isMultiApply, menuItem):
+    '''
+    Apply the given schema to the prims in the current selection.
+    '''
+    primPaths = [ufe.PathString.string(item.path()) for item in _getUsdItemsInSelection()]
+    if not primPaths:
+        print('Cannot apply schema "%s": no USD prim are currently selected.' % schemaTypeName)
```

Comment:
This string should be in the resource and you should use the cmds.warning or cmds.error (not print)